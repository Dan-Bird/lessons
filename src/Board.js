import React, { useState } from 'react';
import Square from './Square';
import { calculateWinner } from './helpers';

const Board = () => {
  const [squares, setSquares] = useState(Array(9).fill(null));
  const [xIsNext, setXIsNext] = useState(true);
  const winner = calculateWinner(squares);

  const getStatus = () => {
    if (winner) {
      return 'Winner: ' + winner;
    } else {
      return 'Next player: ' + (xIsNext ? 'X' : 'O');
    }
  };

  const handleClick = i => {
    const _squares = squares.slice(0);
    if (calculateWinner(squares) || squares[i]) return;
    _squares[i] = xIsNext ? 'X' : 'O';
    setSquares(_squares);
    setXIsNext(!xIsNext);
  };

  const renderSquare = i => {
    return <Square value={squares[i]} onClick={() => handleClick(i)} />;
  };

  return (
    <div>
      <div className="status">{getStatus()}</div>
      <div className="board-row">
        {renderSquare(0)}
        {renderSquare(1)}
        {renderSquare(2)}
      </div>
      <div className="board-row">
        {renderSquare(3)}
        {renderSquare(4)}
        {renderSquare(5)}
      </div>
      <div className="board-row">
        {renderSquare(6)}
        {renderSquare(7)}
        {renderSquare(8)}
      </div>
    </div>
  );
};

export default Board;
