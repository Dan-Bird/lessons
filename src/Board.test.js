import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Board from './Board';

it('matches snapshot', () => {
  const tree = renderer.create(<Board />).toJSON();
  expect(tree).toMatchSnapshot();
});
