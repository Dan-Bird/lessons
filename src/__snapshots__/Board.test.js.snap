// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="status"
  >
    Next player: X
  </div>
  <div
    className="board-row"
  >
    <button
      className="square"
      data-testid="square"
      onClick={[Function]}
    />
    <button
      className="square"
      data-testid="square"
      onClick={[Function]}
    />
    <button
      className="square"
      data-testid="square"
      onClick={[Function]}
    />
  </div>
  <div
    className="board-row"
  >
    <button
      className="square"
      data-testid="square"
      onClick={[Function]}
    />
    <button
      className="square"
      data-testid="square"
      onClick={[Function]}
    />
    <button
      className="square"
      data-testid="square"
      onClick={[Function]}
    />
  </div>
  <div
    className="board-row"
  >
    <button
      className="square"
      data-testid="square"
      onClick={[Function]}
    />
    <button
      className="square"
      data-testid="square"
      onClick={[Function]}
    />
    <button
      className="square"
      data-testid="square"
      onClick={[Function]}
    />
  </div>
</div>
`;
